state.go

```diff
@@ -299,7 +299,17 @@ func (s *State) MemberAdd(member *Member) error {
 		members[member.User.ID] = member
 		guild.Members = append(guild.Members, member)
 	} else {
-		*m = *member // Update the actual data, which will also update the member pointer in the slice
```

Comment:
Could you implement this in the same way as we do in Guild, it will mean that we don't have to update 2 places whenever the Member struct changes, ie:

```
// We are about to replace `m` in the state with `member`, but first we need to
// make sure we preserve any fields that the `member` doesn't contain from `m`.
if member.JoinedAt == "" {
  member.JoinedAt = m.JoinedAt
}
*m = *member
```

Reply:
That looks like a better solution; will do.